README.md

```diff
@@ -0,0 +1,101 @@
+# volpiano-display-utilities
+
+This repository provides a set of modules for the syllabification of latin chant texts and the alignment of chant texts with volpiano according to the conventions of Cantus Database. 
+
+The respository consists of three main utilities:
+1. A tool (`cantus_text_syllabification.syllabify_text`) that parses text and syllabifies it for display with a melody. This tool assumes that input text conforms to Cantus Database specifications for text entry, including handling of incipits, text division, and missing text or music.
+2. A tool (`latin_word_syllabification.syllabify_word`) that splits latin words into syllables based on linguistic rules. This tool could theoretically be used to syllabify latin words from any latin text, whether or not it is transcribed according to Cantus Database specifications. 
+3. A tool (`text_volpiano_alignment.align_text_and_volpiano`) that aligns a chant's text and volpiano together for display in the browser. The tool includes basic handling of misalignments due to encoding error. 
+
+### Text Syllabification
+
+`syllabify_text` parses a chant text encoded to Cantus Database specifications for display with a melody. A brief overview of Cantus Database encoding conventions can be found below or at www.cantusdatabase.org/documents. 
+
+```python
+>>> from cantus_text_syllabification import syllabify_text
+>>> syllabify_text("Angus dei qui {tollis peccata} mundi | Miserere # | ~Agnus")
+'[[["Ag-","nus"], ["de-","i"], ["qui"]], [["{tollis peccata}"]], [["mun-","di"]], [["|"]], [["Mi-","se-","re-","re"], ["#"]], [["|"]], [["~Angus"]]]'
+>>> syllabify_text("Glori@ patr1 &t")
+'ValueError: ...' # Input with invalid characters will return a value error. 
+```
+
+Argument `flatten_result` modifies the function return value.
+
+```python
+>>> syllabify_text("Angus dei qui {tollis peccata} mundi | Miserere # | ~Agnus", flatten_result = True)
+'Ag-nus de-i qui {tollis peccata} mun-di | Mi-se-re-re # | ~Angus'
+```
+
+The function returns a nested list of strings with the following structure:
+ - each element of the list is a sublist that contains the syllabification of a section of the chant text. Sections are created when signaled in the chant text by pipes ("|") or when a substring of the chant text should not be syllabified (eg. when it corresponds to missing music -- these are detailed below)
+ - each section list is a list of words (for sections where syllabification is necessary) or a singleton list for the whole section (for sections where syllabification is not necessary)
```

Comment:
"a singleton list" - I'm not familiar with this - is "singleton list" a technical term for a list with a single item?

Reply:
I don't know if anyone else uses it for lists. I know the term "singleton" from mathematics, where it's very common to mean an collection (usually a set) with only one item...I assumed it was a common word, but maybe it's a technical term of art?